src/System.ValueTuple/src/System/ValueTuple/TupleExtensions.cs

```diff
@@ -0,0 +1,906 @@
+            item2 = value.Item2;
+            item3 = value.Item3;
+            item4 = value.Item4;
+            item5 = value.Item5;
+            item6 = value.Item6;
+            item7 = value.Item7;
+            item8 = value.Rest.Item1;
+            item9 = value.Rest.Item2;
+            item10 = value.Rest.Item3;
+            item11 = value.Rest.Item4;
+            item12 = value.Rest.Item5;
+            item13 = value.Rest.Item6;
+            item14 = value.Rest.Item7;
+            item15 = value.Rest.Rest.Item1;
+            item16 = value.Rest.Rest.Item2;
+            item17 = value.Rest.Rest.Item3;
+            item18 = value.Rest.Rest.Item4;
+            item19 = value.Rest.Rest.Item5;
+        }
+
+        /// <summary>
+        /// Deconstruct a propertly nested <see cref="Tuple"/> with 20 elements.
+        /// </summary>
+        public static void Deconstruct<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T20>(
+            this Tuple<T1, T2, T3, T4, T5, T6, T7, Tuple<T8, T9, T10, T11, T12, T13, T14, Tuple<T15, T16, T17, T18, T19, T20>>> value,
+            out T1 item1, out T2 item2, out T3 item3, out T4 item4, out T5 item5, out T6 item6, out T7 item7, out T8 item8, out T9 item9, out T10 item10, out T11 item11, out T12 item12, out T13 item13, out T14 item14, out T15 item15, out T16 item16, out T17 item17, out T18 item18, out T19 item19, out T20 item20)
+        {
+            item1 = value.Item1;
+            item2 = value.Item2;
+            item3 = value.Item3;
+            item4 = value.Item4;
+            item5 = value.Item5;
+            item6 = value.Item6;
+            item7 = value.Item7;
+            item8 = value.Rest.Item1;
+            item9 = value.Rest.Item2;
+            item10 = value.Rest.Item3;
+            item11 = value.Rest.Item4;
+            item12 = value.Rest.Item5;
+            item13 = value.Rest.Item6;
+            item14 = value.Rest.Item7;
+            item15 = value.Rest.Rest.Item1;
+            item16 = value.Rest.Rest.Item2;
+            item17 = value.Rest.Rest.Item3;
+            item18 = value.Rest.Rest.Item4;
+            item19 = value.Rest.Rest.Item5;
+            item20 = value.Rest.Rest.Item6;
+        }
+
+        /// <summary>
+        /// Deconstruct a propertly nested <see cref="Tuple"/> with 21 elements.
+        /// </summary>
+        public static void Deconstruct<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T20, T21>(
+            this Tuple<T1, T2, T3, T4, T5, T6, T7, Tuple<T8, T9, T10, T11, T12, T13, T14, Tuple<T15, T16, T17, T18, T19, T20, T21>>> value,
+            out T1 item1, out T2 item2, out T3 item3, out T4 item4, out T5 item5, out T6 item6, out T7 item7, out T8 item8, out T9 item9, out T10 item10, out T11 item11, out T12 item12, out T13 item13, out T14 item14, out T15 item15, out T16 item16, out T17 item17, out T18 item18, out T19 item19, out T20 item20, out T21 item21)
```

Comment:
Note that "generate the equivalent of these methods" would not work well. If there are methods, then one can bind them to delegate, so where the methods would live? It would need to be a common location per program (not per assembly).

The other viable alternative is to make compiler intrinsically familiar with System.Tuple and not require Deconstruct methods in these scenarios per language spec. That is a half way to supporting System.Tuple as a secondary tuple compatible type and seems too heavy.